src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.h

```diff
@@ -149,6 +166,60 @@ class GenericThreadStackManagerImpl_OpenThread
                                         const otSrpClientService * aRemovedServices, void * aContext);
     static void OnSrpClientStateChange(const otSockAddr * aServerSockAddr, void * aContext);
 
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD_DNS_CLIENT
+
+    struct DnsClient
```

Comment:
Like I commented elsewhere, I think we should think if it wouldn't be possible to allocate everything on stack in case of the DNS client side.

Reply:
Agree, but in that case creating DnsClient struct has no point, as it will contain plenty of unused fields in a specific callback. I will try to refactor it and create separate structures for resolve and browse then.